src/components/KTableView/KTableView.vue

```diff
@@ -880,6 +1026,75 @@ watch(hasColumnVisibilityMenu, (newVal) => {
     columnVisibility.value = props.tablePreferences.columnVisibility || {}
   }
 }, { immediate: true })
+
+const bulkActionsAll = ref<boolean>(false)
+
+const isBulkActionsIndeterminate = computed((): boolean => {
+  const selectableRows = tableData.value.filter((row) => getRowBulkActionEnabled(row))
+
+  return !!selectableRows.filter((row) => row.selected).length && !!selectableRows.filter((row) => !row.selected).length
+})
+
+const handleIndeterminateChange = (value: boolean) => {
+  if (value) {
+    tableData.value = [...tableData.value].map((row) => ({ ...row, selected: getRowBulkActionEnabled(row) }))
+  } else {
+    // unselect and reset all
+    tableData.value = [...props.data].map((row) => ({ ...row, selected: false }))
+  }
+}
```

Comment:
Please add comments here for the maps, etc.

Reply:
Done